pkg/controller/disruption/disruption.go

```diff
@@ -227,8 +229,9 @@ func (dc *DisruptionController) getPodStatefulSet(controllerRef *metav1.OwnerRef
 
 // getPodDeployments finds deployments for any replicasets which are being managed by deployments.
 func (dc *DisruptionController) getPodDeployment(controllerRef *metav1.OwnerReference, namespace string) (*controllerAndScale, error) {
-	if controllerRef.Kind != controllerKindRS.Kind {
-		return nil, nil
+	ok, err := verifyGroupKind(controllerRef, controllerKindRS.Kind, []string{"apps", "extensions"})
+	if !ok || err != nil {
+		return nil, err
 	}
 	rs, err := dc.rsLister.ReplicaSets(namespace).Get(controllerRef.Name)
 	if err != nil {
```

Comment:
pre-existing, but the same group check ("apps" or "extensions") should be done when resolving the deployment controllerref below. up to you whether you do that here or in a follow-up

Reply:
Updated the PR to include the group test on the deployment controller ref.